master/internal/api_experiment.go

```diff
@@ -235,26 +237,54 @@ func (a *apiServer) deleteExperiment(exp *model.Experiment, user *model.User) er
 	return nil
 }
 
+func protoStateDBCaseString(
+	enumToValue map[string]int32, colName, serializedName, trimFromPrefix string,
+) string {
+	query := fmt.Sprintf("CASE %s::text ", colName)
+	for enum, v := range enumToValue {
+		query += fmt.Sprintf("WHEN '%s' THEN %d ", strings.TrimPrefix(enum, trimFromPrefix), v)
+	}
+	return query + fmt.Sprintf("END AS %s", serializedName)
+}
+
 func (a *apiServer) GetExperiments(
-	_ context.Context, req *apiv1.GetExperimentsRequest,
+	ctx context.Context, req *apiv1.GetExperimentsRequest,
 ) (*apiv1.GetExperimentsResponse, error) {
-	// Construct the experiment filtering expression.
-	var allStates []string
-	for _, state := range req.States {
-		allStates = append(allStates, strings.TrimPrefix(state.String(), "STATE_"))
-	}
-	stateFilterExpr := strings.Join(allStates, ",")
-	userFilterExpr := strings.Join(req.Users, ",")
-	userIds := make([]string, 0)
-	for _, userID := range req.UserIds {
-		userIds = append(userIds, strconv.Itoa(int(userID)))
-	}
-	userIDFilterExpr := strings.Join(userIds, ",")
-	labelFilterExpr := strings.Join(req.Labels, ",")
-	archivedExpr := ""
-	if req.Archived != nil {
-		archivedExpr = strconv.FormatBool(req.Archived.Value)
-	}
+	resp := &apiv1.GetExperimentsResponse{Experiments: []*experimentv1.Experiment{}}
+	query := db.Bun().NewSelect().
+		Model(&resp.Experiments).
+		ModelTableExpr("experiments as e").
+		Column("e.id").
+		ColumnExpr("e.config->>'description' AS description").
+		ColumnExpr("e.config->>'labels' AS labels").
+		ColumnExpr("proto_time(e.start_time) AS start_time").
+		ColumnExpr("proto_time(e.end_time) AS end_time").
+		ColumnExpr(protoStateDBCaseString(experimentv1.State_value, "e.state", "state", "STATE_")).
```

Comment:
thoughts on converting this to bun meaning: make a better `type Experiment struct` bun model type and a `.ToProto` for a lot of this to protoing in the query. as opposed to just using bun as a query builder?

Reply:
Ok tried converting to using more of the Bun ORM features. Conflicted if it is even better.

For experiment config specifically we need to serialize into ```expconf``` then we deserialize into JSON bytes to then serialize into a proto ```struct``` which we then serialize to json where the webui / CLI will deserialize it again.

It is significantly slower than going directly to proto (~80% for 25 experiments ~500% for 2500 experiments). Which might be fine if it provides enough value. 

Beyond that it is just a little awkward at times. For ordering by trial ID count I need to basically query trial IDs once for bun and once for the order statement since Bun splits getting Trial IDs into its own query.

Reply:
> (~450% for 25 experiments ~500% for 2500 experiments)

what precisely is this comparing (response times for the entire endpoint?) and it is 20 -> 100 or like 100 -> 500 ms?

Reply:
response time for the proto function (so ignoring the proto -> json API layer / network) 

50 experiments -> 4ms vs 21ms
250 experiments -> 17ms vs 93ms
2500 experiments -> 151ms vs 929ms

I don't want to like prematurely optimize here but just somewhat concerned that this change could cause a noticeable slowdown on someones cluster (especially since the benchmark I did had no trials and a relatively minimal experiment config)

Going directly proto just using bun as a query builder avoids this but does repeat a lot of the mistakes that ```QueryProtof``` has so I don't know.

Reply:
Talked offline to take the approach of using bun just a query builder for this case

Follow up ticket
https://determinedai.atlassian.net/browse/DET-8203